static/js/services/select2-search.js

```diff
@@ -88,17 +89,44 @@ angular.module('inboxServices').factory('Select2Search',
           });
       };
 
-      var resolveInitialValue = function(selectEl) {
+      var resolveInitialValue = function(selectEl, initialValues) {
+        // TODO: add support for multiples values here
+        initialValues = initialValues[0];
+
+        if (initialValues) {
+          if (!selectEl.children('option[value="' + initialValues + '"]').length) {
+            selectEl.append($('<option value="' + initialValues + '"/>'));
+          }
+          selectEl.val(initialValues);
+        } else {
+          selectEl.val('');
+        }
+
+        var valueResolved;
         var value = selectEl.val();
-        if (!value || value.length === 0) {
-          return $q.resolve(selectEl);
+        if (!(value && value.length)) {
+          valueResolved = $q.resolve();
+        } else {
+          // TODO: add support for multiples values here
+          value = value[0];
+          if (value.indexOf('+') === 0) {
```

Comment:
I'm not sure we can rely on all phone numbers starting with a `+` especially in legacy systems. You could use libphonenumber to validate the string.